src/PrestaShopBundle/Form/Admin/Improve/International/Tax/TaxType.php

```diff
@@ -0,0 +1,130 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Admin\Improve\International\Tax;
+
+use PrestaShopBundle\Form\Admin\Type\SwitchType;
+use PrestaShopBundle\Form\Admin\Type\TranslatableType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\Regex;
+use Symfony\Component\Validator\Constraints\Type;
+
+/**
+ * Form type for tax add/edit
+ */
+class TaxType extends AbstractType
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('name', TranslatableType::class, [
+                'options' => [
+                    'constraints' => [
+                        new Regex([
+                            'pattern' => '/^[^<>={}]*$/u',
+                            'message' => $this->translator->trans(
+                                '%s is invalid.',
+                                [
+                                    sprintf('"%s"', $this->translator->trans('Name', [], 'Admin.Global')),
+                                ],
+                                'Admin.Notifications.Error'
+                            ),
+                        ]),
+                        new NotBlank([
+                            'message' => $this->translator->trans(
+                                'The %s field is required.',
+                                [
+                                    sprintf('"%s"', $this->translator->trans('Name', [], 'Admin.Global')),
+                                ],
+                                'Admin.Notifications.Error'
+                            ),
+                        ]),
+                        new Length([
+                            'max' => 32,
+                            'maxMessage' => $this->translator->trans(
+                                'This field cannot be longer than %limit% characters',
```

Comment:
Maybe it'd be nicer to turn it into a more positive wording, like _This field should have %limit% characters or less_? @TristanLDD?

Reply:
This string was used in CustomerType and possibly somewhere else

Reply:
One question: is this wording necessarily displayed? or does it appear only when the number of characters reaches its limit?

Reply:
It appears as an error when form is submitted with more than 32 characters in this case.

Reply:
Alright, thanks @zuk3975, let's keep your suggestion then: _This field cannot be longer than %limit% characters_ in _Admin.Notifications.Error_.